dev/dev-env-setup.sh

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env bash
+
+set +exv
+
+showHelp() {
+cat << EOF
+Usage: ./install-dev-env.sh [-d] [directory to install virtual environment] [-v] [-q] [-f]
+Development environment setup script for Python in linux-based Operationg Systems (including OSX).
+Note: this script will not work on Windows.
+This script will:
+
+  - Install pyenv if not installed
+  - Retrieve the appropriate Python version (minimum required) for compatibility support
+  - Check if the virtual environment already exists
+    - If it does, prompt for replacement
+      - If replacing, delete old virtual environment.
+  - Create a virtual environment using the minimum required Python version based on previous step logic
+  - Activate the environment
+  - Install required dependencies for the dev envrionment
+
+  Example usage:
+
+  From root of MLflow repository on local with a destination virtualenv path of <MLFLOW_HOME>/.venvs/mlflow-dev:
+
+  dev/dev-env-setup.sh -d $(pwd)/.venvs/mlflow-dev
+
+  Note: it is recommended to preface virtualenv locations with a directory name prefaced by '.' (i.e., ".venvs").
+
+  The default environment setup is for basic functionality, installing the minimum development requirements dependencies.
+  To install the full development environment that supports working on all flavors and running all tests locally, set
+  the flag '-f' or '--full'
+
+-h, -help,        --help        Display help
+
+-d, -directory    --directory   The path to install the virtual environment into
```

Comment:
Does `./dev/dev-env-setup.sh -directory` or `./dev/dev-env-setup.sh --directory` work?

Reply:
it does now!

Reply:
btw do we need `-directory`?

Reply:
removed those redundant arg keys